src/pages/index.js

```diff
@@ -78,24 +126,123 @@ addcreateCardButton.addEventListener("click", () => {
   formValidators["add-button-form"].resetValidation();
 });
 
+avatarImage.addEventListener("click", () => {
+  avatarEditModal.open();
+});
+
 // EVENT HANDLERS
 
+function handleLikeCard(cardId) {
+  return api
+    .likeCard(cardId)
+    .then(() => {
+      const card = cardsSection.getItem(cardId);
+      if (card) {
+        card._updateLikeStatus(true);
+      }
+    })
+    .catch((err) => {
+      console.error(`Error liking card: ${err}`);
+    });
+}
+
+function handleUnlikeCard(cardId) {
+  return api
+    .unlikeCard(cardId)
+    .then(() => {
+      const card = cardsSection.getItem(cardId);
+      if (card) {
+        card._updateLikeStatus(false);
+      }
+    })
+    .catch((err) => {
+      console.error(`Error unliking card: ${err}`);
+    });
+}
+
+function handleDeleteButtonClick(cardId) {
+  deleteCardModal.open(cardId);
+}
+
 function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo({
+  profileSaveButton.textContent = "Saving...";
+
+  const updatedUserInfo = {
     name: inputValues.name,
-    job: inputValues.job,
-  });
+    about: inputValues.about,
+  };
 
-  profileModal.close();
+  api
+    .updateUserInfo(updatedUserInfo)
+    .then((updatedData) => {
+      userInfo.setUserInfo(updatedData);
+      profileModal.close();
+    })
+    .catch((err) => {
+      console.error("Error updating user information:", err);
+    })
+    .finally(() => {
+      profileSaveButton.textContent = "Save";
+    });
 }
 
 function handleAddCardSubmit(inputValues) {
+  addCardSaveButton.textContent = "Saving...";
   const name = inputValues.title;
   const link = inputValues.image;
   const data = { name, link };
 
-  cardsSection.addItem(createCard(data));
-  cardModal.close();
+  api
+    .createCard(data)
+    .then((newCard) => {
+      cardsSection.addItem(createCard(newCard));
+      cardModal.close();
+    })
+    .catch((err) => {
+      console.error("Error creating card:", err);
+    })
+    .finally(() => {
+      addCardSaveButton.textContent = "Create";
+    });
+}
+
+function handleDeleteCard(cardId) {
+  api
+    .deleteCard(cardId)
+    .then(() => {
+      const cardElement = document.querySelector(`[data-id='${cardId}']`);
+      if (cardElement) {
+        cardElement.remove();
+      }
+    })
+    .catch((err) => {
+      console.error("Error deleting card:", err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors